kernel/src/log_segment.rs

```diff
@@ -514,6 +514,51 @@ pub(crate) struct ListedLogFiles {
     pub latest_crc_file: Option<ParsedLogPath>,
 }
 
+impl ListedLogFiles {
+    fn new(
+        ascending_commit_files: Vec<ParsedLogPath>,
+        ascending_compaction_files: Vec<ParsedLogPath>,
+        checkpoint_parts: Vec<ParsedLogPath>,
+        latest_crc_file: Option<ParsedLogPath>,
+    ) -> Self {
+        // We are adding debug_asserts here, so they don't impact the runtime performance of the released binaries
+        debug_assert!(ascending_compaction_files.windows(2).all(|pair| {
+            let [first, second] = pair else {
+                unreachable!()
+            };
+            match (&first.file_type, &second.file_type) {
+                (
+                    LogPathFileType::CompactedCommit { hi: hi0 },
+                    LogPathFileType::CompactedCommit { hi: hi1 },
+                ) => first.version <= second.version && *hi0 <= *hi1,
+                _ => false,
+            }
+        }));
+        debug_assert!(checkpoint_parts.iter().all(|p| p.is_checkpoint()));
+        debug_assert!(checkpoint_parts
+            .windows(2)
+            .all(|p| p[0].version == p[1].version));
+        debug_assert!(
```

Comment:
maybe a comment to help readers?
```suggestion
        // check that only one of the following exists: (1) a singular `SinglePartCheckpoint` or
        // `UUID` checkpoint or (2) a multi-part, and all parts are there and nothing else
        debug_assert!(
```